File: download_images.py
from duckduckgo_search import DDGS
import requests
import os

vehicles = [
    {"name": "scoopy", "query": "Honda Scoopy transparent background png"},
    {"name": "highlander", "query": "Toyota Highlander transparent background png"},
    {"name": "ranger", "query": "Ford Ranger transparent background png"},
    {"name": "alphard", "query": "Toyota Alphard transparent background png"},
    {"name": "tuktuk", "query": "Cambodia Tuk Tuk transparent background png"}
]

ddgs = DDGS()
os.makedirs("public", exist_ok=True)

for v in vehicles:
    print(f"Searching for {v['name']}...")
    try:
        results = list(ddgs.images(
            v["query"],
            region="wt-wt",
            safesearch="moderate",
            size="Large",
            color="Transparent",
            max_results=3
        ))
        for res in results:
            url = res['image']
            try:
                print(f"Downloading {url}")
                img_data = requests.get(url, timeout=5).content
                with open(f"public/{v['name']}.png", "wb") as f:
                    f.write(img_data)
                print(f"Saved {v['name']}.png")
                break
            except Exception as e:
                print(f"Failed to download: {e}")
    except Exception as e:
        print(f"Error searching: {e}")

File: backend/src/scripts/downloadImages.js
const { image_search } = require('duckduckgo-images-api');
const fs = require('fs');
const path = require('path');
const https = require('https');

const vehicles = [
  { name: 'scoopy', query: 'Honda Scoopy white background isolated profile' },
  { name: 'highlander', query: 'Toyota Highlander white background isolated profile' },
  { name: 'ranger', query: 'Ford Ranger white background isolated profile' },
  { name: 'alphard', query: 'Toyota Alphard white background isolated profile' },
  { name: 'tuktuk', query: 'Cambodian Tuk Tuk white background isolated' }
];

async function download(url, dest) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(dest);
    https.get(url, (response) => {
      response.pipe(file);
      file.on('finish', () => {
        file.close();
        resolve();
      });
    }).on('error', (err) => {
      fs.unlink(dest, () => {});
      reject(err);
    });
  });
}

async function run() {
  for (const v of vehicles) {
    try {
      console.log(`Searching for ${v.query}...`);
      const results = await image_search({ query: v.query, moderate: true });
      if (results && results.length > 0) {
        let downloaded = false;
        for (let i = 0; i < Math.min(5, results.length); i++) {
          try {
            const dest = path.join(__dirname, `../../../../public/${v.name}.jpg`);
            await download(results[i].image, dest);
            console.log(`Successfully downloaded ${v.name}`);
            downloaded = true;
            break;
          } catch (e) {
            console.log(`Failed to download result ${i}, trying next...`);
          }
        }
        if (!downloaded) console.log(`Failed to download any image for ${v.name}`);
      }
    } catch (e) {
      console.error(`Error searching ${v.name}:`, e.message);
    }
  }
}

run();
